### PR TITLE
Fixed #29860 -- Allowed BaseValidator to accept a callable limit_value.

### PR DESCRIPTION
**Pull request by kingbuzzman**
https://code.djangoproject.com/ticket/29860

### CODE REVIEW

File: tests/migrations/test_autodetector.py [hunk 1]
```diff
@@ -1304,6 +1307,33 @@ def test_different_regex_does_alter(self):
...
+                ('dob', models.DateTimeField(validators=[MinValueValidator(datetime.datetime.now)])),
+            ]
+        )
+        to_state = ModelState(
+            "testapp", "model", [('dob', models.DateTimeField(validators=[lt_today_validator]))]
+        )
+        changes = self.get_changes([from_state], [to_state])
+        self.assertNumberMigrations(changes, "testapp", 1)
+        self.assertOperationTypes(changes, "testapp", 0, ["AlterField"])
+
```
Comment (charettes) [inline tests/migrations/test_autodetector.py]: I'm not sure there's much value in adding these two tests as they were really specific to the `SimpleLazyObject` issue. They pass with and without the changes to `BaseValidator.__call__`.

Comment (kingbuzzman) [reply tests/migrations/test_autodetector.py]: I had the same thought -- just wasnt too sure. Either way, its gone

File: docs/ref/validators.txt [hunk 2]
```diff
@@ -235,32 +235,60 @@ to, or in lieu of custom ``field.clean()`` methods.
...
 
+    :param limit_value: Value or callable.
+    :param message: If not ``None``, overrides message.
+
     Raises a :exc:`~django.core.exceptions.ValidationError` with a code of
     ``'max_value'`` if ``value`` is greater than ``limit_value``.
+    
+    .. versionchanged:: 2.2
+    
+        limit_value could be a callable.
```
Comment (charettes) [inline docs/ref/validators.txt]: Double backtip `limit_value` and I think that _Added support for callable `limit_value` would be more appropriate? I'm not a native speaker myself so I'll let one chime in.

File: docs/releases/2.2.txt [hunk 3]
```diff
@@ -251,7 +251,9 @@ URLs
 Validators
 ~~~~~~~~~~
 
-* ...
+* ``BaseValidator`` now supports a callable to be passed in in order to lazy
```
Comment (charettes) [inline docs/releases/2.2.txt]: `in in`

File: docs/releases/2.2.txt [hunk 4]
```diff
@@ -251,7 +251,9 @@ URLs
 Validators
 ~~~~~~~~~~
 
-* ...
+* ``BaseValidator`` now supports a callable to be passed in in order to lazy 
+  evaluate the validator value. This helps when validating dates that
```
Comment (charettes) [inline docs/releases/2.2.txt]: `limit value`?

**Comment by hvdklauw** [comment]
This does not work when used with model fields: https://code.djangoproject.com/ticket/30328

**Pull request merged**